packages/web-components/fast-foundation/src/badge/badge.ts

```diff
@@ -36,4 +36,21 @@ export class Badge extends FASTElement {
      */
     @attr({ mode: "boolean" })
     public circular: boolean;
+
+    public generateBadgeStyle = () => {
+        const fill: string = `background-color: var(--badge-fill-${this.fill});`;
+        const color: string = `color: var(--badge-color-${this.color});`;
+
+        if (!this.fill && !this.color) {
```

Comment:
You should move this early-exit condition to the top so we don't allocate variables if they aren't needed.

Reply:
moved 👍 - thanks!